src/messages/MOSDOp.h

```diff
@@ -371,6 +370,7 @@ struct ceph_osd_request_head {
       encode(flags, payload);
       encode(reqid, payload);
       encode_trace(payload, features);
+      encode_otel_trace(payload, features);
```

Comment:
There's the conversation about the encoding breaking things, and this here is the fundamental issue — you just shoved a new data member into the middle of an encoding, which is an *absolutely* incompatible change, and didn't do anything to make sure all the various endpoints involved were able to keep communicating.

I'm guessing this was modeled after 6159027fdac0, but that looks to have been riding along on an existing check for the SERVER_LUMINOUS flag to keep things okay (or else it was broken and fixed up elsewhere), which this PR doesn't get to take advantage of.

Reply:
yes. will post a similar fix that checks on the SERVER_REEF flag.

Reply:
created a fix here: https://github.com/ceph/ceph/pull/51288
if fix is acceptable, will squash into one commit